app/scripts/models/auth_brokers/oauth.js

```diff
@@ -179,6 +179,10 @@ define(function (require, exports, module) {
     },
 
     transformLink: function (link) {
+      if (! /^\//.test(link)) {
```

Comment:
You know, I'm so used to using RegExps for everything, I didn't think of the simple way to do it.